### ISSUE TITLE
我们目前担心如果xxl-job-admin宕机后，如何保证 xxl-job-executor 中的job可以正常被触发执行，这个后续咱们有方案涉及嘛

### ISSUE THREAD

**Issue by dxt1218** [xuxueli/xxl-job#3162]
Please answer some questions before submitting your issue. Thanks!
2.2.2-WATSONS

**Comment by coderManFans** [comment]
可以多节点部署，数据库走高可用，基本没啥问题

**Comment by dxt1218** [comment]
> 可以多节点部署，数据库走高可用，基本没啥问题

目前admin还是单节点 ，所以需要保证高可用

**Comment by KevinClair** [comment]
exectuor的任务执行是admin触发的，admin如果单节点部署且故障，任务不会被触发执行，所以可靠的方式依然是集群部署。

**Comment by dxt1218** [comment]
> exectuor的任务执行是admin触发的，admin如果单节点部署且故障，任务不会被触发执行，所以可靠的方式依然是admin集群部署。

这种回复不亚于，“何不食肉糜”。不要脱离问题局限。

**Comment by coderManFans** [comment]
如果这样的话，可以考虑客户端缓存任务下次调度时间，由客户端进行自己调度执行，但是这样的话就要考虑数据一致性了，另外客户端自己调度的话也不能撑太长时间

**Comment by dxt1218** [comment]
> 如果这样的话，可以考虑客户端缓存任务下次调度时间，由客户端进行自己调度执行，但是这样的话就要考虑数据一致性了，另外客户端自己调度的话也不能撑太长时间

是的 其实这个是不错的办法  ，之前考虑的是缓存任务最后一次执行时间来做监控。然后大于间隔的某个阈值后 自我调用，俗称自救。不过一致性的话  首先在exectuor中选择一个作为 可以自我调用的角色。统一由一个服务来自我调度。不过确实不能撑太长时间。

**Comment by Kenwangxr** [comment]
admin可以多节点部署，使用数据库做交互，已验证可用

**Comment by xuxueli** [comment]
> 可用

+1，支持集群部署，实现调度HA
